### PR TITLE
Feat add external progress bar

### PR DESCRIPTION
**Pull request by bragaru-i**
This is a pre-ticket for: https://github.com/Collaborne/backlog/issues/1332
### What id does:
- Create a **ProgressBar** created only from media listeners(can be used **outside the MediaProvider's** context)

Why usage of the  **media listeners—because usage of the [usePlayerContext](https://github.com/Collaborne/video-player/blob/main/src/hooks/use-player-context.ts) gives us a `RefObject` to the `MediaStore`, and we can get fresh data from there via listening to it and not causing rerenderings

### CODE REVIEW

File: src/components/progress-bar/EventProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import { ProgressBarStyled } from './components/ProgressBarStyled';
+
+interface EventProgressBarProps extends SliderProps {
+	mediaListener?: EmitterListeners;
+	setCurrentTime?: (relativeSeconds: number) => void;
+	highlights?: Highlight[];
+	getHighlightColorBlended?: BlendColors;
+}
+
+/** A MUI Slider configured for displaying currentTime/duration values from `MediaStore`
```
Comment (ronnyroeller) [inline src/components/progress-bar/EventProgressBar.tsx:21]: Add newline to put `/**` on separate line

File: src/components/progress-bar/EventProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * uses `EmitterListeners` for displaying data
+ * @category React Component
+ * @category UI Controls
+ */
+export const EventProgressBar: FC<EventProgressBarProps> = ({
```
Comment (ronnyroeller) [inline src/components/progress-bar/EventProgressBar.tsx:26]: Can we find a better name for the component? We have now `ProgressBar` and `EventProgressBar`, both with very similar comment to describe it. I would find it really hard to decide when to use what, without reading the details of the code.

**Review by ronnyroeller — APPROVED** [review]
LGTM (pls address comments before merging)

File: src/components/progress-bar/EventProgressBar.tsx (hunk 2 above)
Comment (bragaru-i) [reply src/components/progress-bar/EventProgressBar.tsx:26]: @ronnyroeller, maybe `EventBasedProgressBar` ?

**Pull request merged**